### PR TITLE
libobs: Fix mixed wide and narrow characters

### PR DESCRIPTION
**Pull request by GkvJwa**
### Description

Fix mixed wide and narrow characters on windows
vsnprintf cannot format wide and narrow at the same time
### Motivation and Context

The log is found in the windows environment as follows:
12:37:44.371: Sec. Software Status:
12:37:44.372: 
12:37:44.372: 
12:37:44.373: Current Date/Time: 2022-11-10, 12:37:44
Then I read the code and found that vsnprintf goes to format wide characters and utf8 strings at the same time
### How Has This Been Tested?

Checked that the log analyser continues to correctly read the Windows version
### Types of changes

Bug fix (non-breaking change which fixes an issue)
### Checklist:

- [ ] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [ ] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [ ] My code is not on the master branch.
- [ ] The code has been tested.
- [ ] All commit messages are properly formatted and commits squashed where appropriate.
- [ ] I have included updates to all appropriate documentation.

### CODE REVIEW

File: libobs/obs-windows.c [hunk 1]
```diff
@@ -316,8 +316,23 @@ static void log_security_products_by_type(IWSCProductList *prod_list, int type)
...
+		size_t utf8_name_length =
+			os_wcs_to_utf8(name, name_length, NULL, 0);
+
+		if (utf8_name_length) {
+			utf8_name = bmalloc(utf8_name_length);
+			os_wcs_to_utf8(name, name_length, utf8_name,
+				       utf8_name_length);
+		}
+
+		if (utf8_name) {
```
Comment (notr1ch) [inline libobs/obs-windows.c:330]: This null check seems unnecessary since the code above already writes to the pointer without checking.

Comment (GkvJwa) [reply libobs/obs-windows.c:328]: review again

Comment (notr1ch) [reply libobs/obs-windows.c:328]: The check can be removed entirely as `bmalloc` will never return NULL.

Comment (GkvJwa) [reply libobs/obs-windows.c]: > The check can be removed entirely as `bmalloc` will never return NULL.
review again

Comment (GkvJwa) [reply libobs/obs-windows.c]: > The check can be removed entirely as `bmalloc` will never return NULL.

review?

**Review by gxalpha — CHANGES_REQUESTED** [review]
Please make sure to always squish your commits back together into one commit after making changes.

Also please show some patience, most people here are volunteers doing this in their free time. You may need to wait a few days (in some cases even weeks) for someone to re-review.

**Pull request closed without merging**